umake/frameworks/baseinstaller.py

```diff
@@ -199,18 +203,23 @@ def get_metadata_and_check_license(self, result):
             if url is None or (self.checksum_type and checksum is None):
                 logger.error("Download page changed its syntax or is not parsable")
                 UI.return_main_screen()
-            self.download_requests.append(DownloadItem(url, Checksum(self.checksum_type, checksum)))
 
-            if license_txt.getvalue() != "":
-                logger.debug("Check license agreement.")
-                UI.display(LicenseAgreement(strip_tags(license_txt.getvalue()).strip(),
-                                            self.start_download_and_install,
-                                            UI.return_main_screen))
-            elif self.expect_license and not self.auto_accept_license:
-                logger.error("We were expecting to find a license on the download page, we didn't.")
-                UI.return_main_screen()
-            else:
-                self.start_download_and_install()
```

Comment:
A flake8 complaint about method complexity. I would've wanted to reduce it further but I couldn't.

I suppose this will do for now.